### PR TITLE
feat(analytics): add nft and native token activity endpoints

### LINKED ISSUE #15: Tracking Issue - `Analytics`
There are several statistics that can help us better understand how a network is used. The following analytic endpoints should be supported by Chronicle:

* [x] #328
  * [x] Total number of minted NFTs
  * [x] Total number of native tokens created
  * [x] #424
* [ ] #329
  * [x] #419
  * [x] #423
  * [x] #516

The activity analytics should be implemented for set time intervals of 24h, 7d, 30d, and 90d.

In addition to these required statistics @Alexandcoats has some more brainstorming ideas below, which we could tackle afterwards.

Here is what the endpoints will look like:

#### Activity

* [ ] `/activity ?startIndex & endIndex`
  * [x] `/blocks` (total count)
    * [x] `/transaction`
    * [x] `/tagged-data`
    * [x] `/milestone`
    * [x] `/treasury-transaction`
  * [x] `/outputs` (total count/amount created
    * [x] `/alias`
    * [x] `/basic`
    * [x] `/nft`
    * [x] `/foundry`
    * [x] `/storage-deposit` 
  * [x] `/base-tokens` (use ledger update table)
  * [ ] `/native-tokens` (use ledger update table)
  * [ ] `/nfts` (use ledger update table)

#### Ledger Status

* [x] `/ledger ?ledger_index` (total count unspent)
  * [x] `/storage-deposit` (byte cost at ledger index)
  * [x] `/nfts` (count of the number of unspent nft ouputs)
  * [x] `/native-tokens` (count of the number of unspent foundry outputs)

### LINKED ISSUE #329: Add activity statistics within time intervals

### LINKED ISSUE #516: Total number of NFTs / native tokens moved (minted / transfered / burned)
NFTs

Let set `I` be all distinct NFT IDs of unspent outputs at ledger index _i_.
Let set `I_n` be all distinct NFT IDs of unspent outputs that were created between milestones _i_ and _i+n_.

* How many NFTs were created b/n milestone _i_ and _i+1_? (`countCreated`)
  * use `count setDifference(I_n, I)`
* How many NFTs were transfered in milestone _i_? (`countTransfered`)
  * use `count setIntersection(I, I_n)`
* How many NFTs were burned in milestone _i_? (`countBurned`)
  * use `count setDifference(I, I_n)`

Native Tokens

### PR DESCRIPTION
**Pull request by Alexandcoats**
Closes #516.
Closes #329.
Closes #15.

### CODE REVIEW

File: bin/inx-chronicle/src/api/stardust/analytics/routes.rs [hunk 1]
```diff
@@ -38,6 +38,8 @@ pub fn routes() -> Router {
             "/activity",
             Router::new()
                 .route("/addresses", get(address_analytics))
+                .route("/native-tokens", get(foundry_output_analytics))
+                .route("/nfts", get(nft_output_analytics))
```
Comment (grtlr) [inline bin/inx-chronicle/src/api/stardust/analytics/routes.rs:42]: Please rename the methods in `get(...)` to `*-activity-analytics`. We should do this analogous for the `*-ledger-analytics`.

File: src/db/collections/outputs/mod.rs [hunk 2]
```diff
@@ -468,6 +468,127 @@ impl MongoDb {
...
+        end_index: Option<MilestoneIndex>,
+    ) -> Result<OutputDiffAnalyticsResult, Error> {
+        Ok(self
+            .db
+            .collection::<OutputDiffAnalyticsResult>(OutputDocument::COLLECTION)
+            .aggregate(
+                vec![
+                    doc! { "$match": {
+                        "output.kind": "nft",
+                        "metadata.booked.milestone_index": { "$not": { "$gt": start_index } },
```
Comment (grtlr) [inline src/db/collections/outputs/mod.rs:492]: Booked is always there, so can't we use `$lte`?

File: src/db/collections/outputs/mod.rs [hunk 3]
```diff
@@ -468,6 +468,127 @@ impl MongoDb {
...
+        end_index: Option<MilestoneIndex>,
+    ) -> Result<OutputDiffAnalyticsResult, Error> {
+        Ok(self
+            .db
+            .collection::<OutputDiffAnalyticsResult>(OutputDocument::COLLECTION)
+            .aggregate(
+                vec![
+                    doc! { "$match": {
+                        "output.kind": "foundry",
+                        "metadata.booked.milestone_index": { "$not": { "$gt": start_index } },
```
Comment (grtlr) [inline src/db/collections/outputs/mod.rs:548]: see above

File: src/db/collections/outputs/mod.rs [hunk 4]
```diff
@@ -468,6 +468,127 @@ impl MongoDb {
...
+                                    { "metadata.spent_metadata.spent.milestone_index": { "$not": { "$lte": start_index } } },
+                                ],
+                            } },
+                            { "$project": {
+                                "foundry_id": "$output.foundry_id"
+                            } },
+                        ],
+                        "end_state": [
+                            { "$match": {
+                                "metadata.booked.milestone_index": { "$not": { "$lte": start_index } },
```
Comment (grtlr) [inline src/db/collections/outputs/mod.rs:564]: see above

File: src/db/collections/outputs/mod.rs [hunk 5]
```diff
@@ -468,6 +468,127 @@ impl MongoDb {
...
+                        "metadata.booked.milestone_index": { "$not": { "$gt": start_index } },
+                    } },
+                    doc! { "$facet": {
+                        "start_state": [
+                            { "$match": {
+                                "$or": [
+                                    { "metadata.spent_metadata.spent": null },
+                                    { "metadata.spent_metadata.spent.milestone_index": { "$not": { "$lte": start_index } } },
+                                ],
+                            } },
```
Comment (grtlr) [inline src/db/collections/outputs/mod.rs:557]: In the spirit of #342, we should create descriptive functions for these. That would make reading the queries much simpler.

File: src/db/collections/outputs/mod.rs (hunk 2 above)
Comment (Alexandcoats) [reply src/db/collections/outputs/mod.rs:492]: This is not because booked can be null, but because `start_index` can.

**Review by grtlr — APPROVED** [review]

**Pull request merged**